### ISSUE TITLE
story(issue-218): evaluate a fine-tuned model against a held-out set

### ISSUE THREAD

**Issue by Zaba505** [z5labs/devex#231]
### Description

`Training.Evaluate` (landed with #222) runs `lstmeval` against the *training*
set, so what it reports is how well the model fit the data it was shown — not
how it does on data it has not seen. Those are different numbers, and the
second one is the one that decides whether a fine-tune is worth shipping. A
model that memorised forty lines reports a beautiful BCER and is worthless.

Splitting the ground truth is the job. `tesstrain` does it with a ratio and a
shuffle; the shape that fits this module better is a second directory, because
a caller who cares about generalisation usually already knows which pages they
held back — and a ratio hides a random split behind a number, which makes the
result irreproducible run to run.

```go
func (tr *Training) WithEvaluationSet(source *dagger.Directory) *Training
```

The samples for it are built by the machinery that already exists — the same
box generation, the same `lstm.train` pass — and land in their own list file,
so the training run is unchanged and only `lstmeval`'s `--eval_listfile`
moves. Unset, `Evaluate` keeps reporting the training-set error it does today,
with the doc comment saying which one it is.

### Acceptance Criteria

- [ ] `WithEvaluationSet` takes a directory of image + `.gt.txt` pairs, validated by the same rules the training set is
- [ ] `Evaluate` reports against it when set, and against the training set when not
- [ ] One test trains on part of a set and evaluates on the rest, asserting the two rates differ
- [ ] Pairs in the evaluation set never reach `lstmtraining`'s `--train_listfile`
- [ ] `dagger develop` re-run in module and tests, generated code committed

### Related Issues

Builds on #218. Deferred from #222.
